rf_tests.go

```diff
@@ -342,13 +342,20 @@ func deleteRFML(c cliContext) error {
 	return nil
 }
 
-// uploadRFML is a wrapper around test creating/updating functions
-func uploadRFML(c cliContext, api rfmlAPI) error {
+// uploadTests is a wrapper around test creating/updating functions
+func uploadRFTests(c cliContext, api rfAPI) error {
 	if c.Bool("synchronous-upload") {
 		rfmlUploadConcurrency = 1
 	}
 	if path := c.Args().First(); path != "" {
-		err := uploadSingleRFMLFile(path)
+		var err error
+
+		if strings.HasSuffix(path, ".rfml") {
```

Comment:
It'd be nice to determine the file type by it's contents rather than file extension? Could we try reading them as WISP and if that fails fall back to RFML (or vice versa)?

Reply:
@jbarber I've decided to leave this as is. `rfml` already expects the extension to be correct in a few places, but my main issue is that just a simple fallback can potentially hide an error with the file.

Eg:
- Try to upload invalid wisp
- Get error wisp is invalid
- Fallback to try to upload rfml
- Get error rfml is invalid
- return rfml invalid error to user for an invalid wisp

The argument there would be to add better error handling, but that's a bit more involved than I would like for this PR.
If we want that, I would like to do it in a separate PR and revisit how rfml is handled as well.

The other thing you mentioned is just changing the extensions to `.wisp`, that's a simple change to make now.

Reply:
I think we covered this in our 1:1 - but to recap, the backend is the source of truth, so I think it's ok to delegate validation of WISP to it and that means the workflow would be:
* Try to parse as JSON, on failure parse as RFML (or vice versa)
* Do stuff with all the parsed files

IMO it's fine to ignore in this PR.

Reply:
I'll ignore it for now. My main issue is that I believe we will be providing a misleading error in some circumstances(like the one I described above).

I will address this in a cleanup PR with a proposed solution and we can discuss further there.